plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoSession.java

```diff
@@ -368,9 +368,14 @@ private MongoTable loadTableSchema(SchemaTableName schemaTableName)
             MongoColumnHandle columnHandle = buildColumnHandle(columnMetadata);
             columnHandles.add(columnHandle);
         }
+        List<MongoColumnHandle> gatheredColumnHandles = columnHandles.build();
+
+        if (!caseInsensitiveNameMatching && gatheredColumnHandles.isEmpty()) {
+            throw new TableNotFoundException(schemaTableName, format("Table '%s.%s' not found", schemaTableName.getTableName(), schemaTableName.getTableName()), null);
+        }
```

Comment:
These changes look wrong. It would be better to check the collection's existence and throw an exception if it doesn't exist.

Reply:
@ebyhr , Thanks to review! 

I would like to hear your opinion before modifying the code.

`collectionExists` method is not using `caseInsensitiveNameMatching`. So even if `caseInsensitiveNameMatching` is `false`, comparison ignores case sensitivity and `tbl == TBL` returns true.
Therefore, I am going to modify this method to consider `caseInsensitiveNameMatching` and return the result.

Afterwards, i will add code to check whether the table actually exists by calling `collectionExists` in `loadTableSchema.

This implementation seems to make two changes in situations where `caseInsensitiveNameMatching` is `false`.

1. Before change it, Documents were added to `_schema` despite not being able to utilize tables containing uppercase letters, but they are no longer added.
2. Throws a TableNotFoundException if the table contains uppercase characters, even if the table has already been added to `_schema`.

Reply:
In addition, If `caseInsensitiveMatching=false`, does it mean that Trino will handle only the table composed of lowercase letters in MongoDB?

Otherwise, users will get strange results when:

1. A Collection consisting of uppercase letters called TEST exists in MongoDB.
2. The test table is confirmed in show tables in mongodb.test_db
3. TableNotFoundException when searching select * from mongodb.test_db.test

If `caseInsensitiveMatching=false`, wouldn't it make sense to make all functions work only for lowercase tables in MongoDB?

Reply:
Finally, I have an opinion about `toRemoteTableName`.

If the user has created three collections, 'Test', 'tesT', and 'tEst', the method below is unlikely to be deterministic because it returns only the first candidate.

```
    private String toRemoteSchemaName(String schemaName)
    {
        verify(schemaName.equals(schemaName.toLowerCase(ENGLISH)), "schemaName not in lower-case: %s", schemaName);
        if (!caseInsensitiveNameMatching) {
            return schemaName;
        }
        for (String remoteSchemaName : listDatabaseNames()) {
            if (schemaName.equals(remoteSchemaName.toLowerCase(ENGLISH))) { <-- Here
                return remoteSchemaName;
            }
        }
        return schemaName;
    }
```
 Wouldn't it be better to throw an Ambigious exception when there are multiple candidates in the `caseInsensitiveMatching=true` situation?

Reply:
> Wouldn't it be better to throw an Ambigious exception when there are multiple candidates in the caseInsensitiveMatching=true situation?

It would be better to throw such exception. You can refer to BigQuery connector implementation.